devdocs/troubleshooting-guide.md

```diff
@@ -0,0 +1,214 @@
+
+7. **ERROR**: A handful of failed test cases (<10).
+
+  **SOLUTION**: Re-run the failed tests with TestNG, all test cases should pass eventually (it may take a few runs). If there are tests that persistently fail and not addressed in other parts of this guide, you may [request for help in the issue tracker](https://github.com/TEAMMATES/teammates/issues/new).
+  
+8. **ERROR**: Browser alert about invalid SSL certificate.
+
+  **SOLUTION**: Ensure that test.app.url in your test.properties uses -dot- instead of '.' when using secondary subdomains.
+  Eg. `http://4-30-dot-teammates-john.appspot.com`
+9. **ERROR**: Test cases failed due to accented characters.
+
+  **SOLUTION**: Ensure that the text file encoding for your Eclipse workspace has been set to UTF-8 as specified under Prerequisites.
+10. **ERROR**: Error message in the console about "incorrect date format".
+
+  **SOLUTION**: Ensure the date format of your computer matches the below. For Windows, [this link](http://www.sevenforums.com/tutorials/3530-time-format-change.html) may be useful.
+
+  ![troubleshooting-test-5.png](images/troubleshooting-test-5.png)
+
+11. **ERROR**: InstructorCourseEnrollPageUiTest fails due to a hash mismatch.
+
+  **REASON**: The hash of the file in the repo is different from the one used in the live site due to line endings are encoded on different Operating Systems.
+
+  **SOLUTION**: Replace "/src/main/webapp/files/Course Enroll Sample Spreadsheet.csv" with [this copy of file](https://teammatesv4.appspot.com/files/Course%20Enroll%20Sample%20Spreadsheet.csv).
+
+12. **ERROR**: "no chromedrive_osx" when testing on Mac using Chrome.
+
+  **SOLUTION**: Download the file from [here]( http://chromedriver.storage.googleapis.com/index.html) and extract to /src/test/resources/lib/selenium, then rename it by adding "`_osx`". Refresh the folder in eclipse.
+
+13. **ERROR**:java.lang.UnsupportedClassVersionError when running on staging server.
+
+  **SOLUTION**: This can happen if you have Java 8 installed on your computer. Change Eclipse configuration to use Java 7 instead. Here is a useful [reference]( http://java.wildstartech.com/Java-Platform-Standard-Edition/mac-os-x-java-development/how-to-configure-eclipse-to-run-with-java-7-when-java-8-is-installed).
+  
+
+14. **ERROR**: InstructorCourseDetailsPageUiTest fails in production server due to Gmail's access restriction (e.g javax.mail.AuthenticationFailedException)
+
+  **SOLUTION**: "Allow access for less secure apps" for the accounts used in that particular test. As doing so will leave the accounts more vulnerable to security issues, it is strongly recommended that the access is revoked after the test passes. Check for [reference]( https://support.google.com/accounts/answer/6010255?hl=en).
+
+#### Errors specific to Linux
```

Comment:
Instead of adding a new section, we can just make this a new number, e.g:
```
15. **ERROR (on Linux)**: ...
```
This is to be consistent with number 4 up there.